### PR TITLE
[TRB-39717] Stitching volumes from AWS to kubeturbo fails for manually created volumes

### PR DESCRIPTION
**Pull request by tian-ma**
https://jsw.ibm.com/browse/TRB-39717
For AWS:
Original code requires that a volume ID is in the format `aws://us-east-2c/vol-0e4eaa3ef79bcb5a9`

This PR aims to remove this restriction as discussed in the bug report.
Try to extract volume id by parsing Labels if the volume ID doesn't explicitly contain the zone identifier

Before the fix is applied:
The volume "stitching-bug-validation" appears twice and total number of volumes is 4

After the fix is applied:
Total number of volumes is 3

image

image

### CODE REVIEW

File: pkg/discovery/stitching/volume_stitching_manager.go [hunk 1]
```diff
@@ -198,16 +198,41 @@ func (aws *awsVolumeUUIDGetter) GetVolumeUUID(vol *api.PersistentVolume) (string
...
-	suffix := volID[len(awsVolPrefix):]
+	suffix := volID
+	if strings.HasPrefix(volID, awsVolPrefix) {
+		suffix = volID[len(awsVolPrefix):]
+	}
+
 	parts := strings.Split(suffix, "/")
-	if len(parts) != 2 {
-		return "", fmt.Errorf("failed to split uuid (%d): %v, for volume %v", len(parts), parts, volName)
+	if len(parts) < 2 {
```
Comment (irfanurrehman) [inline pkg/discovery/stitching/volume_stitching_manager.go:214]: Line 246 below can cause a problem if this condition is true.
`parts[1]` does not seem to be set (nil pointer dereference) (`len(parts) < 2`)

**Review by irfanurrehman — APPROVED** [review]
Thanks for doing this.
Please squash the commits to those with more meaningful messages before merging.

File: pkg/discovery/stitching/volume_stitching_manager.go [hunk 2]
```diff
@@ -198,16 +198,41 @@ func (aws *awsVolumeUUIDGetter) GetVolumeUUID(vol *api.PersistentVolume) (string
...
+				if strings.HasSuffix(labelKey, "zone") {
+					zone = labelV
+				}
+			}
+			if zone != "" {
+				parts[0] = zone
+			} else if region != "" {
+				parts[0] = region + "-"
+			}
+		}
```
Comment (ading1977) [inline pkg/discovery/stitching/volume_stitching_manager.go:235]: Nit:

Remove the `else` brackets.

Comment (ading1977) [reply pkg/discovery/stitching/volume_stitching_manager.go:235]: Also can you put an example of the volume labels?

**Review by ading1977 — APPROVED** [review]

**Pull request merged**